demos/video-filtering.c

```diff
@@ -65,8 +65,8 @@
 
 // Stuff that would be common to each API
 
-void *init();
-void uninit(void *priv);
+static void *init(void);
+static void uninit(void *priv);
```

Comment:
I'm not sure if I agree with this commit. I understand that it is not normal for a file local to a function to be marked as static, and I also understand why the compiler warns you for this and that it's annoying - but I think not labelling these functions as static illustrates more clearly that these are functions that would normally be defined by a header file rather than as part of the same source file.

(Also, neither gcc nor clang seem to warn about it for me)

Reply:
There is no warning. I didn't know these functions are meant to be public. I thought they are only forward declarations, so internal linkage was the issue what I considered.

Reply:
Okay, then I think I'll keep this functions "public", if only to illustrate the example better. Maybe we could split it up into separate files and have those as part of a .h or something?